src/llmq/ehf_signals.cpp

```diff
@@ -54,8 +54,15 @@ void CEHFSignalsHandler::UpdatedBlockTip(const CBlockIndex* const pindexNew)
         // TODO: v20 will never attempt to create EHF messages on main net; if this is needed it will be done by v20.1 or v21 nodes
         return;
     }
-    // TODO: should do this for all not-yet-signied bits
-    trySignEHFSignal(Params().GetConsensus().vDeployments[Consensus::DEPLOYMENT_MN_RR].bit, pindexNew);
+    const auto ehfSignals = WITH_LOCK(cs_main, return chainstate.GetMNHFSignalsStage(pindexNew));
```

Comment:
Why are we locking cs_main here?

Reply:
ok I think this can be removed since GetMNHFSignalsStage will lock cs_main internally when needed